packages/atomic/src/components/facets/atomic-segmented-facet/atomic-segmented-facet.tsx

```diff
@@ -0,0 +1,146 @@
+import {Component, Element, h, Prop, State, VNode} from '@stencil/core';
+import {
+  Bindings,
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../utils/initialization-utils';
+import {BaseFacet} from '../facet-common';
+import {
+  buildFacet,
+  buildSearchStatus,
+  Facet,
+  FacetOptions,
+  FacetSortCriterion,
+  FacetState,
+  FacetValue,
+  SearchStatus,
+  SearchStatusState,
+} from '@coveo/headless';
+import {registerFacetToStore} from '../../../utils/store';
+import {getFieldValueCaption} from '../../../utils/field-utils';
+import {FacetValueLabelHighlight} from '../facet-value-label-highlight/facet-value-label-highlight';
+import {FacetValueBox} from '../facet-value-box/facet-value-box';
+
+/**
+ * @internal
+ */
+@Component({
+  tag: 'atomic-segmented-facet',
+  styleUrl: 'atomic-segmented-facet.pcss',
+  shadow: true,
+})
+export class AtomicSegmentedFacet
+  implements InitializableComponent, BaseFacet<Facet, FacetState>
+{
+  @InitializeBindings() public bindings!: Bindings;
+  public searchStatus!: SearchStatus;
+  @State()
+  public searchStatusState!: SearchStatusState;
+  @BindStateToController('facet')
+  @State()
+  public facetState!: FacetState;
+  public facet!: Facet;
+  @Element() private host!: HTMLElement;
+  @State() public error!: Error;
+
+  /**
+   * Specifies a unique identifier for the facet.
+   */
+  @Prop({mutable: true, reflect: true}) public facetId?: string;
+  /**
+   * The field whose values you want to display in the facet.
+   */
+  @Prop({reflect: true}) public field!: string;
+  /**
+   * The non-localized label for the facet.
+   */
+  @Prop({reflect: true}) public label = 'no-label';
+  /**
+   * Whether to exclude the parents of folded results when estimating the result count for each facet value.
+   */
+  @Prop({reflect: true}) public filterFacetCount = true;
+  /**
+   * The maximum number of results to scan in the index to ensure that the facet lists all potential facet values.
+   * Note: A high injectionDepth may negatively impact the facet request performance.
+   * Minimum: `0`
+   */
+  @Prop() public injectionDepth = 1000;
```

Comment:
So... we have a lot of "facets components", which expose the same option, but no interface to make sure all facets are consistent with one another.

That's a problem, because I'm fairly certain we'll forget to apply options changes eventually (we've certainly had that problem in JSUI, with inconsistent options between components).

This can stay like this for this PR (it's outside of the scope of your story), just pointing out the problem. I'll try to work on a solution.

Reply:
No, just that all facets components should use the same wording for all options, and that if one changes, all components should be affected at the same time.

See this PR for explanation: https://github.com/coveo/ui-kit/pull/2013

Reply:
I see ! Very cool